### PR TITLE
Expose the Fold.Tee module for general use

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: src/Streamly/Data/Fold/Tee.hs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+-- Copyright   : (c) 2021 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- The 'Tee' type is a newtype wrapper over the 'Streamly.Data.Fold.Fold' type providing
+-- distributive 'Applicative', 'Semigroup', 'Monoid', 'Num', 'Floating' and
+-- 'Fractional' instances. The input received by the composed 'Tee' is
+-- replicated and distributed to both the constituent Tee's.
```
Comment (adithyaov) [inline src/Streamly/Data/Fold/Tee.hs:12]:
```suggestion
-- replicated and distributed to both the constituent Tees.
```

Or `Tee-s`?

**Review by adithyaov — APPROVED** [review]
We also update the changelog for new features right?

File: src/Streamly/Data/Fold/Tee.hs (hunk 1 above)
Comment (harendra-kumar) [reply src/Streamly/Data/Fold/Tee.hs:12]: Will change.

**Comment by harendra-kumar** [comment]
> We also update the changelog for new features right?

Added to changelog.

**Pull request merged**